src/configuration.cc

```diff
@@ -26,6 +26,13 @@
 namespace google {
 
 namespace {
+#ifndef AGENT_VERSION
+#define AGENT_VERSION 0.0
+#endif
+
+#define STRINGIFY_H(x) #x
+#define STRINGIFY(x) STRINGIFY_H(x)
```

Comment:
why are two macro definitions required? Isn't the first one sufficient?

Reply:
Macros are substituted outside-in. If we simply called `STRINGIFY_H(AGENT_VERSION)`, the result would have been the string `"AGENT_VERSION"` (because `#x` would be evaluated with `x` set to `AGENT_VERSION`). Adding another macro layer will expand both macros in the first pass, so `STRINGIFY(AGENT_VERSION)` expands to `STRINGIFY_H(0.0.17-1)` (assuming `AGENT_VERSION` is set to `0.0.17-1`), which then expands on the next pass to `#x` with `x` set to `0.0.17-1`, resulting in the string `"0.0.17-1"`. The C preprocessor is weird.

Reply:
thanks for the explanation. Please add a unit test for it to make this more explicit.

Reply:
I actually tried. Doing this right would require rebuilding `configuration.cc` with a different `AGENT_VERSION` value (changing the way we build files under test), which is more upheaval than this PR warrants.
I suppose I could pull the `STRINGIFY` macro out into one of the headers (`format.h`?) and test it separately... Let me know if you want me to do that.

Reply:
As discussed offline, referenced the GCC docs instead.